extensions/prototypes/labgraph_monitor/src/contexts/WSContext/WSContext.tsx

```diff
@@ -0,0 +1,72 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+import { ReactNode, useRef, createContext, useEffect } from 'react';
+import { w3cwebsocket as W3CWebSocket } from 'websocket';
+import { RootState } from '../../redux/store';
+import { useSelector, useDispatch } from 'react-redux';
+import { setConnection, setGraph } from '../../redux/reducers/ws/WSReducer';
+import { copyRealtimeGraph } from '../../redux/reducers/mock/mockReducer';
+import WS_STATE from '../../redux/reducers/ws/enums/WS_STATE';
+import startStreamRequest from './json/startStreamRequest.json';
+import endStreamRequest from './json/endStreamRequest.json';
+import _ from 'lodash';
+
+const GraphContext = createContext<{}>({});
+
+/**
+ * A context component used to share WebSocket related information with different component.
+ * All interactions with the LabGraph WebSockets API are happening inside this context.
+ *
+ * @param {ReactNode} props represents the react components wrapped by this context
+ * @returns {JSX.Element}
+ */
+const WSContextProvider: React.FC<ReactNode> = ({ children }): JSX.Element => {
+    const { connection, graph } = useSelector((state: RootState) => state.ws);
+
+    const clientRef = useRef<any>(null);
```

Comment:
nitpick, not helpful to have `any`, could we replace that with its actual type?

Reply:
Done